core/src/main/java/org/elasticsearch/index/shard/ShardSplittingQuery.java

```diff
@@ -84,44 +94,85 @@ public Scorer scorer(LeafReaderContext context) throws IOException {
                         Uid.decodeId(ref.bytes, ref.offset, ref.length), null);
                     return shardId == targetShardId;
                 };
-                if (terms == null) { // this is the common case - no partitioning and no _routing values
+                if (terms == null) {
+                    // this is the common case - no partitioning and no _routing values
+                    // in this case we also don't do anything special with regards to nested docs since we basically delete
+                    // by ID and parent and nested all have the same id.
                     assert indexMetaData.isRoutingPartitionedIndex() == false;
                     findSplitDocs(IdFieldMapper.NAME, includeInShard, leafReader, bitSet::set);
                 } else {
+                    final BitSet parentBitSet;
+                    final IntPredicate includeDoc;
```

Comment:
So just to double check, the `includeDoc` is to ensure that only root docs get selected and later we select the nested docs of the selected root docs in `markChildDocs(...)`, right?

Reply:
correct